### ISSUE TITLE
Add locale parameter for translations in templates

### ISSUE THREAD

**Issue by bennyborn** [contao/contao#4014]
### Description

With #4013 we'll now be able to request translations for different languages to do something like this:

```php
dump( $this->translator->trans('MSC.target.0', [], 'contao_default','de') ); // "In neuem Fenster öffnen"
dump( $this->translator->trans('MSC.target.0', [], 'contao_default','en') ); // "Open in new window"
dump( $this->translator->trans('MSC.target.0', [], 'contao_default','fr') ); // "Ouvrir dans une nouvelle fenêtre"
dump( $this->translator->trans('MSC.target.0', [], 'contao_default','it') ); // "Si apre in una nuova finestra"
```

This is all fine for modules and controllers using the Translator service.

If we need translations of different languages within a template like this...

```php
dump( $this->trans('MSC.target.0', [], 'contao_default','de') );
dump( $this->trans('MSC.target.0', [], 'contao_default','en') );
dump( $this->trans('MSC.target.0', [], 'contao_default','fr') );
dump( $this->trans('MSC.target.0', [], 'contao_default','it') );
```
... it won't work because the trans method of the Template class lacks the `$locale` parameter

https://github.com/contao/contao/blob/9e0dceaa6660feddeb36608035ad79cd6e6ba136/core-bundle/src/Resources/contao/library/Contao/Template.php#L382-L394

With the Twig-Support in the works shouldn't there be something possible like this?

```twig
{{ 'MSC.target.0'|trans([],'contao_default','de') }}
{{ 'MSC.target.0'|trans([],'contao_default','en') }}
{{ 'MSC.target.0'|trans([],'contao_default','fr') }}
{{ 'MSC.target.0'|trans([],'contao_default','it') }}
```
> Dicslaimer: I haven't worked with Twig that much so there is a good chance my syntax is total BS. Maybe @m-vo could give it a look? 👀

Long story short: Having the "core" translation stuff fixed with #4013 I'd appreciate if we could also make sure that different lanuages work for the Twig integration as well as using the trans method of the Template class.

**Comment by fritzmg** [comment]
> With the Twig-Support in the works shouldn't there be something possible like this?

That should already work :). Syntax looks correct.

**Comment by m-vo** [comment]
Explicitly setting a locale works like you posted.

But there is an ongoing discussion (since 2020…) about how a locale could be forced when rendering a template (and on other places): https://github.com/symfony/symfony/issues/35925

**Comment by fritzmg** [comment]
Keep in mind though that changing the locale for the legacy translations can be "dangerous". You will need to make sure that you reset it to the initial locale, otherwise all following translations might be in the wrong language.

**Comment by bennyborn** [comment]
> Keep in mind though that changing the locale for the legacy translations can be "dangerous". You will need to make sure that you reset it to the initial locale, otherwise all following translations might be in the wrong language.

Prior to @ausi's PR that is correct but I just tested it and after forcing a specific language all following translations (without an explicit locale specified) still use the "global locale".

```php
dump( $this->translator->trans('MSC.target.0', [], 'contao_default') ); // "In neuem Fenster öffnen"
dump( $this->translator->trans('MSC.target.0', [], 'contao_default','de') ); // "In neuem Fenster öffnen"
dump( $this->translator->trans('MSC.target.0', [], 'contao_default','en') ); // "Open in new window"
dump( $this->translator->trans('MSC.target.0', [], 'contao_default','fr') ); // "Ouvrir dans une nouvelle fenêtre"
dump( $this->translator->trans('MSC.target.0', [], 'contao_default','it') ); // "Si apre in una nuova finestra"
dump( $this->translator->trans('MSC.target.0', [], 'contao_default') ); // "In neuem Fenster öffnen"
```

**Comment by ausi** [comment]
`translator->trans()` should always work correctly. But if you use `translator->trans(…, 'fr')` and after that access `$GLOBALS['TL_LANG'][…]` directly (as the core does in many places) it would be in the wrong locale.

**Comment by fabpot** [comment]
See https://github.com/symfony/symfony/pull/45793 (which will be part of Symfony 6.1).
